exchanges/bybit/bybit_test.go

```diff
@@ -85,3423 +78,3120 @@ func TestGetKlines(t *testing.T) {
+	if mockTests {
+		t.Skip("skipping authenticated function for mock testing")
+	}
 	sharedtestvalues.SkipTestIfCredentialsUnset(t, b)
-
-	_, err := b.GetWalletFundRecords(context.Background(), "2021-09-11", "2021-10-09", "ETH", "", "", 0, 0)
+	_, err := b.GetOpenOrders(context.Background(), "", "", "", "", "", "", "", "", 0, 100)
+	if !errors.Is(err, errCategoryNotSet) {
+		t.Fatalf("expected %v, got %v", errCategoryNotSet, err)
+	}
+	_, err = b.GetOpenOrders(context.Background(), "spot", "", "", "", "", "", "", "", 0, 0)
 	if err != nil {
 		t.Error(err)
 	}
 }
 
-func TestGetWalletWithdrawalRecords(t *testing.T) {
+func TestCancelAllTradeOrders(t *testing.T) {
 	t.Parallel()
-	sharedtestvalues.SkipTestIfCredentialsUnset(t, b)
-
-	_, err := b.GetWalletWithdrawalRecords(context.Background(), "2021-09-11", "2021-10-09", "", currency.ETH, 0, 0)
+	if mockTests {
+		t.Skip("skipping authenticated function for mock testing")
+	}
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, b, canManipulateRealOrders)
+	_, err := b.CancelAllTradeOrders(context.Background(), nil)
+	if !errors.Is(err, errNilArgument) {
+		t.Fatalf("expected %v, got %v", errNilArgument, err)
+	}
+	_, err = b.CancelAllTradeOrders(context.Background(), &CancelAllOrdersParam{})
+	if !errors.Is(err, errCategoryNotSet) {
+		t.Fatalf("expected %v, got %v", errCategoryNotSet, err)
+	}
+	_, err = b.CancelAllTradeOrders(context.Background(), &CancelAllOrdersParam{Category: "option"})
 	if err != nil {
 		t.Error(err)
 	}
 }
 
-func TestGetAssetExchangeRecords(t *testing.T) {
+func TestGetTradeOrderHistory(t *testing.T) {
 	t.Parallel()
+	if mockTests {
+		t.Skip("skipping authenticated function for mock testing")
+	}
 	sharedtestvalues.SkipTestIfCredentialsUnset(t, b)
-
-	_, err := b.GetAssetExchangeRecords(context.Background(), "", 0, 0)
+	start := time.Now().Add(-time.Hour * 24 * 6)
+	end := time.Now()
+	_, err := b.GetTradeOrderHistory(context.Background(), "", "", "", "", "", "", "", "", "", start, end, 100)
+	if !errors.Is(err, errCategoryNotSet) {
+		t.Fatalf("expected %v, got %v", errCategoryNotSet, err)
+	}
+	_, err = b.GetTradeOrderHistory(context.Background(), "spot", spotTradablePair.String(), "", "", "BTC", "", "StopOrder", "", "", start, end, 100)
 	if err != nil {
 		t.Error(err)
 	}
 }
 
-// test cases for USDTMarginedFutures
-
-func TestGetUSDTFuturesKlineData(t *testing.T) {
+func TestPlaceBatchOrder(t *testing.T) {
 	t.Parallel()
-	pair, err := currency.NewPairFromString("BTCUSDT")
+	if mockTests {
+		t.Skip("skipping authenticated function for mock testing")
+	}
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, b, canManipulateRealOrders)
+	_, err := b.PlaceBatchOrder(context.Background(), nil)
+	if !errors.Is(err, errNilArgument) {
+		t.Fatalf("expected %v, got %v", errNilArgument, err)
+	}
+	_, err = b.PlaceBatchOrder(context.Background(), &PlaceBatchOrderParam{})
+	if !errors.Is(err, errCategoryNotSet) {
+		t.Fatalf("expected %v, got %v", errCategoryNotSet, err)
```

Comment:
`expected category not set, got invalid category, only 'option' and 'linear' categories are allowed`